src/main/java/jenkins/branch/DeadBranchStrategyDescriptor.java

```diff
@@ -53,12 +53,9 @@ public boolean isApplicable(java.lang.Class<? extends MultiBranchProject> jobTyp
     @SuppressWarnings("unused") // used by stapler
     public static List<DeadBranchStrategyDescriptor> forProject(Class<? extends MultiBranchProject> jobType) {
         List<DeadBranchStrategyDescriptor> result = new ArrayList<DeadBranchStrategyDescriptor>();
-        for (Descriptor<DeadBranchStrategy> d : Jenkins.getInstance().getDescriptorList(DeadBranchStrategy.class)) {
-            if (d instanceof DeadBranchStrategyDescriptor) {
-                DeadBranchStrategyDescriptor descriptor = (DeadBranchStrategyDescriptor) d;
-                if (descriptor.isApplicable(jobType)) {
-                    result.add(descriptor);
-                }
+        for (DeadBranchStrategyDescriptor descriptor : ExtensionList.lookup(DeadBranchStrategyDescriptor.class)) {
```

Comment:
Unless I am very much mistaken this will instantiate a new parallel extension list of singletons and there will be two DeadBranchStrategyDescriptor "singletons", one stored in `Jenkins.descriptorLists` from the boot sequence processing and the other stored in `extensionLists` resulting from this lookup.